sqlx-core/src/error.rs

```diff
@@ -192,6 +209,29 @@ pub trait DatabaseError: 'static + Send + Sync + StdError {
     fn constraint(&self) -> Option<&str> {
         None
     }
+
+    /// Returns the kind of the error, if supported.
+    ///
+    /// ### Note
+    /// Not all back-ends behave the same when reporting the error code.
+    fn kind(&self) -> Option<ErrorKind> {
```

Comment:
Can you explain your reasoning for choosing `Option<ErrorKind>` over, say, having an `Other` (or maybe `Unspecified`) variant in the `ErrorKind` enum?

I guess I'm just concerned that `kind` returning `None` might imply that errors can sometimes... not be errors?

Reply:
I think I would also prefer this to not have a default implementation just to ensure that each database driver must at least _choose_ a kind to return.

Reply:
> Can you explain your reasoning for choosing Option<ErrorKind> over, say, having an Other (or maybe Unspecified) variant in the ErrorKind enum?

Reply:
> Can you explain your reasoning for choosing Option<ErrorKind> over, say, having an Other (or maybe Unspecified) variant in the ErrorKind enum?

Being completely honest, I don't remember it accurately anymore, but I guess I didn't think of creating an `Other` variant?  But having an `Other` variant makes much more sense indeed, I'll apply the changes

Reply:
Thanks!